include/proteus/JitEngineDevice.hpp

```diff
@@ -489,6 +491,16 @@ void JitEngineDevice<ImplT>::specializeIR(Module &M, StringRef FnName,
         ArrayRef<RuntimeConstant>{RC,
                                   static_cast<size_t>(NumRuntimeConstants)});
 
+  if (!getJitVariableMap().empty()) {
+    for (auto &F : M.getFunctionList()) {
+      if (auto OptionalRCVec = matchJitVariableMap(F.getName())) {
+        TransformLambdaSpecialization::transform(M, F, OptionalRCVec.value());
+        getJitVariableMap().erase(F.getName());
```

Comment:
Better to keep iterator from the match and use that for erase, rather than use the key?

Reply:
Avoid searching, right?